addons/mail/models/mail_thread.py

```diff
@@ -2274,6 +2274,28 @@ def _message_compute_parent_id(self, parent_id):
                 parent_id = current_ancestor.parent_id.id if current_ancestor.parent_id else parent_id
         return parent_id
 
+    def _message_compute_subject(self, record_names=None, model_description=None):
+        """ Helper to get the default subject when creating a message on this thread
+            For most records it should just be the name of the record,
+            override in child model when this does not make sense
+            :param record_name: {thread_id -> record_name} for records which were already computed
+            :return: {thread_id -> record_name} for every records
+        """
+        record_names = dict(record_names) if record_names else dict()
+
+        missing_names = self.filtered(lambda rec: not record_names.get(rec.id)).name_get()
+        record_name_gets = {key: value for key, value in missing_names if value}
+        record_names.update(record_name_gets)
+
+        # if any name could not be found at all, query the name of the model and use that by default
+        name_is_missing = any(filter(lambda thread_id: not record_name_gets.get(thread_id),
```

Comment:
Actually, can we have a missing name at all ? As you call name_get for records not in record_names, how will we have missing names ? Or is it only if we have a void string as name_get ?

Reply:
And if we keep it, could be simplified with a list comprehension based on self + record_name_gets (+ record_names) I think ^^

Reply:
hm... yeah actually at worst it would be an empty string. This simplifies a bunch of stuff ^^'